core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -131,6 +132,7 @@ protected void configure() {
             String type = randomAsciiOfLengthBetween(1, 10);
             mapperService.merge(type, new CompressedXContent(PutMappingRequest.buildFromSimplifiedDef(type,
                     DATE_FIELD_NAME, "type=date",
+                    DATE_FIELD_NAME_REGEX, "type=date,format=YYYY\\-MM",
```

Comment:
i did try that but got failures when it gets to handle regex on dates. The accepted date format need all special regex parameters to be escaped, so the regex (here just a plain date) could be valid.